typespec/Applications.Core/environments.tsp

```diff
@@ -103,24 +110,47 @@ model SecretConfig {
   secret?: string;
 }
 
-@doc("The Cloud providers configuration")
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("ProviderConfigProperties specifies provider configuration details needed for recipes.")
+model ProviderConfigProperties extends Record<unknown> {
+  @doc("The secrets for the referenced provider resource.")
+  secrets?: Record<ProviderSecret>;
+}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("EnvironmentVariables describes structure enabling environment variables to be set.")
+model EnvironmentVariables extends Record<unknown> {
```

Comment:
This should be EnvironmentVariables extends Record<**string**>

Reply:
This is what I get when I use `Record<string>`: **Type 'Record<Applications.Core.RecipeSecret>' is not assignable to type 'string'**.

I also see that this is what we did for Extenders: https://github.com/radius-project/radius/blob/main/typespec/Applications.Core/extenders.tsp#L56.

Reply:
@ytimocin, i'm suggesting towards being more type strict in this case (extenders and provider config need 'unknown' as their values which is not the case here) ..here's a draft. check if this may work
```
model AdditionalProperties extends Record<string> {}

@doc("EnvironmentVariables describes structure enabling environment variables to be set.")
model EnvironmentVariables {
  ... AdditionalProperties;

  secrets?: Record<RecipeSecret>;
}

@doc("Specifies the secret details for the provider.")
model RecipeSecret {
  @doc("The resource id for the secret store containing credentials.")
  source: string;

  @doc("The key for the secret in the secret store.")
  key: string;
}
```

Reply:
Discussed offline